pkg/ring/model.go

```diff
@@ -210,7 +210,11 @@ func (d *Desc) mergeWithTime(mergeable memberlist.Mergeable, localCAS bool, now
 				// We are deleting entry "now", and should not keep old timestamp, because there may already be pending
 				// message in the gossip network with newer timestamp (but still older than "now").
 				// Such message would "resurrect" this deleted entry.
-				ting.Timestamp = now.Unix()
+				// We can have a timestamp more recent than now in some cases (eg. clock skew into the future). For that we need to still use timestamp
+				// forcing the ring not to ignore LEFT state
```

Comment:
Can u reference this line? (if timestamp on the ring is in "the future" we need to use the timestamp of the ring otherwise the gossip message will be ignored by the peers)

https://github.com/cortexproject/cortex/blob/d3f46c53616dcdf5c670f3e3a4d371e44b10683e/pkg/ring/model.go#L196